src/main/groovy/lang/MetaClassImpl.java

```diff
@@ -1832,6 +1832,9 @@ public Object getProperty(Class sender, Object object, String name, boolean useS
             } catch (IllegalArgumentException e) {
                 // can't access the field directly but there may be a getter
                 mp = null;
+            } catch (GroovyRuntimeException e) {
+                // can't access the field directly but there may be a getter
+                mp = null;
```

Comment:
I don't understand why this was added.  Was there a test that failed because of the other changes that required this.  I would have assumed an `AccessControlException` from `CachedField.getProperty` would be treated similar to the `IllegalAccessException` in the same method.